api/database/factories/PoolCandidateFactory.php

```diff
@@ -49,9 +49,8 @@ public function definition()
             'submitted_steps' => array_slice(
                 array_column(ApplicationStep::cases(), 'name'),
                 0,
-                $this->faker->numberBetween(0, count(ApplicationStep::cases()) - 1)
+                $this->faker->numberBetween(0, count(ApplicationStep::cases()) - 2)
```

Comment:
Incomplete applications never have the full set, if the application is complete (see below) always have the full set